File: Olympiad/COCI/official/2008/contest4_solutions/vauvau.cpp
#include <cstdio>

int f( int x, int a, int b ) {
   return (x-1) % (a+b) < a;
}

void solve( int x, int a, int b, int c, int d ) {
   if( f( x, a, b ) + f( x, c, d ) == 0 ) printf( "none\n" );
   if( f( x, a, b ) + f( x, c, d ) == 1 ) printf( "one\n" );
   if( f( x, a, b ) + f( x, c, d ) == 2 ) printf( "both\n" );
}

int main( void ) {
   int A, B, C, D;
   scanf( "%d%d%d%d", &A, &B, &C, &D );

   int P, M, G;
   scanf( "%d%d%d", &P, &M, &G );

   solve( P, A, B, C, D );
   solve( M, A, B, C, D );
   solve( G, A, B, C, D );

   return 0;
}

File: Olympiad/COCI/official/2008/contest4_solutions/vauvau.pas
program vauvau;

function f( x, a, b : longint ) : longint;
begin
   if (x-1) mod (a+b) < a then f := 1 else f := 0;
end;

procedure solve( x, a, b, c, d : longint );
begin
   if f( x, a, b ) + f( x, c, d ) = 0 then writeln( 'none\n' );
   if f( x, a, b ) + f( x, c, d ) = 1 then writeln( 'one\n' );
   if f( x, a, b ) + f( x, c, d ) = 2 then writeln( 'both\n' );
end;

var
   A, B, C, D : longint;
   P, M, G : longint;

begin
   readln( A, B, C, D );
   readln( P, M, G );

   solve( P, A, B, C, D );
   solve( M, A, B, C, D );
   solve( G, A, B, C, D );

end.
